src/App.tsx

```diff
@@ -1,14 +1,56 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FILTER_BY } from './constants';
+
+function getFilterTodos(todos: Todo[], query: string, filterBy: string) {
+  let copy = [...todos];
+
+  if (query) {
+    const lowQuery = query.toLowerCase();
+
+    copy = copy.filter(elem => elem.title.toLowerCase().includes(lowQuery));
+  }
+
+  switch (filterBy) {
+    case FILTER_BY.ACTIVE:
+      return copy.filter(elem => !elem.completed);
+    case FILTER_BY.COMPLETED:
+      return copy.filter(elem => elem.completed);
+    default:
+      return copy;
+  }
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<string>('');
+  const [loading, setLoading] = useState(true);
+  const [todo, setTodo] = useState<Todo | null>();
```

Comment:
The `todo` state is declared as `Todo | null`, but it is not explicitly initialized to `null`. It's a good practice to initialize it to `null` to avoid potential undefined behavior.